bundler/spec/plugins/command_spec.rb

```diff
@@ -29,7 +29,7 @@ def exec(command, args)
   end
 
   it "accepts the arguments" do
-    build_repo2 do
+    update_repo2 do
```

Comment:
Without this change, the spec prints the bug report template with a cryptic error. I'm not sure what's going on there or whether it could affect some real usage of checksums. 

I think the test change makes sense, but this is the error in any case:

```
       Invoking `/Users/deivid/.asdf/installs/ruby/3.2.3/bin/ruby -I/Users/deivid/Code/rubygems/rubygems/bundler/spec -r/Users/deivid/Code/rubygems/rubygems/bundler/spec/support/artifice/compact_index.rb -r/Users/deivid/Code/rubygems/rubygems/bundler/spec/support/hax.rb /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/bin/bundle plugin install the-echoer --source https://gem.repo2` failed with output:
     
       ----------------------------------------------------------------------
       --- ERROR REPORT TEMPLATE -------------------------------------------------------
     
       ```
       TypeError: no implicit conversion of nil into String
         /Users/deivid/.asdf/installs/ruby/3.2.3/lib/ruby/3.2.0/rubygems/package/digest_io.rb:58:in `<<'
                 /Users/deivid/.asdf/installs/ruby/3.2.3/lib/ruby/3.2.0/rubygems/package/digest_io.rb:58:in `block in write'
                 /Users/deivid/.asdf/installs/ruby/3.2.3/lib/ruby/3.2.0/rubygems/package/digest_io.rb:57:in `each'
                 /Users/deivid/.asdf/installs/ruby/3.2.3/lib/ruby/3.2.0/rubygems/package/digest_io.rb:57:in `write'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/compact_index_client/cache_file.rb:106:in `block in append'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/compact_index_client/cache_file.rb:98:in `block (2 levels) in open'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/compact_index_client/cache_file.rb:97:in `open'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/compact_index_client/cache_file.rb:97:in `open'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/compact_index_client/cache_file.rb:97:in `block in open'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/shared_helpers.rb:105:in `filesystem_access'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/compact_index_client/cache_file.rb:96:in `open'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/compact_index_client/cache_file.rb:106:in `append'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/compact_index_client/updater.rb:40:in `block in append'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/compact_index_client/cache_file.rb:34:in `block in copy'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/compact_index_client/cache_file.rb:55:in `initialize'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/compact_index_client/cache_file.rb:25:in `new'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/compact_index_client/cache_file.rb:25:in `copy'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/compact_index_client/updater.rb:29:in `append'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/compact_index_client/updater.rb:17:in `update'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/compact_index_client/cache.rb:80:in `fetch'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/compact_index_client/cache.rb:26:in `versions'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/compact_index_client/parser.rb:54:in `info_checksums'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/compact_index_client/parser.rb:48:in `available?'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/compact_index_client.rb:89:in `available?'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/fetcher/compact_index.rb:55:in `available?'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/fetcher/compact_index.rb:13:in `call'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/fetcher/compact_index.rb:13:in `block in compact_index_request'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/fetcher.rb:232:in `block in fetchers'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/fetcher.rb:232:in `drop_while'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/fetcher.rb:232:in `fetchers'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/fetcher.rb:205:in `api_fetcher?'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/source/rubygems.rb:399:in `select'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/source/rubygems.rb:399:in `api_fetchers'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/source/rubygems.rb:404:in `block in remote_specs'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/index.rb:9:in `build'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/source/rubygems.rb:403:in `remote_specs'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/source/rubygems.rb:138:in `specs'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/resolver.rb:41:in `block in setup_solver'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/resolver.rb:246:in `all_versions_for'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/resolver.rb:54:in `block in setup_solver'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/resolver.rb:342:in `filtered_versions_for'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/resolver.rb:58:in `block in setup_solver'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/resolver.rb:419:in `select_sorted_versions'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/resolver.rb:398:in `block in prepare_dependencies'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/resolver.rb:392:in `each'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/resolver.rb:392:in `map'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/resolver.rb:392:in `prepare_dependencies'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/resolver.rb:63:in `setup_solver'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/resolver.rb:28:in `start'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/definition.rb:626:in `start_resolution'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/definition.rb:305:in `resolve'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/definition.rb:179:in `resolve_remotely!'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/plugin/installer.rb:37:in `install_definition'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/plugin/installer.rb:109:in `block in install_all_sources'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/settings.rb:158:in `temporary'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/plugin/installer.rb:106:in `install_all_sources'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/plugin/installer.rb:98:in `install_rubygems'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/plugin/installer.rb:26:in `install'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/plugin.rb:41:in `install'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/cli/plugin.rb:18:in `install'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/vendor/thor/lib/thor/command.rb:28:in `run'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/vendor/thor/lib/thor.rb:527:in `dispatch'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/vendor/thor/lib/thor/invocation.rb:116:in `invoke'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/vendor/thor/lib/thor.rb:338:in `block in subcommand'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/vendor/thor/lib/thor/command.rb:28:in `run'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/vendor/thor/lib/thor.rb:527:in `dispatch'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/cli.rb:35:in `dispatch'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/vendor/thor/lib/thor/base.rb:584:in `start'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/cli.rb:29:in `start'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/exe/bundle:28:in `block in <top (required)>'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/lib/bundler/friendly_errors.rb:117:in `with_friendly_errors'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/gems/bundler-2.6.0.dev/exe/bundle:20:in `<top (required)>'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/bin/bundle:25:in `load'
                 /Users/deivid/Code/rubygems/rubygems/bundler/tmp/1/gems/system/bin/bundle:25:in `<main>'
```

\cc @martinemde Just in case it rings any bells.